src/accelerate/data_loader.py

```diff
@@ -361,7 +361,15 @@ def __init__(self, dataset, device=None, rng_types=None, generator=None, **kwarg
 
     def __iter__(self):
         if self.rng_types is not None:
-            synchronize_rng_states(self.rng_types, self.generator)
+            batch_sampler = self.sampler if isinstance(self.sampler, BatchSampler) else self.batch_sampler
```

Comment:
See the example script above in the PR, it has various scenarios.

Reply:
see https://github.com/huggingface/accelerate/issues/679, it helps reduce the number of queries

Reply:
https://huggingface.co/docs/datasets/v2.4.0/en/use_with_pytorch#use-a-batchsampler

Reply:
Wow! Thanks a lot!
I spent hours finding a way to query my dataset with a list of indices instead of collating data outside the dataset(so that I can write batching logic in the dataset/reduce the number of queries), and failed(thus I was forced to use DataLoader as a sampler(sharded by accelerate), as mentioned in #624 ).
Now I know I can just use batch sampler as sampler without passing batch size to `DataLoader`.
Thanks!!!😁👍